component/commons/handler/src/main/java/org/wso2/carbon/identity/password/expiry/validation/handler/PasswordExpiryPolicyUtils.java

```diff
@@ -0,0 +1,98 @@
+import org.wso2.carbon.identity.event.IdentityEventConfigBuilder;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.bean.ModuleConfiguration;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementException;
+import org.wso2.carbon.idp.mgt.IdentityProviderManager;
+
+/**
+ * Read password expiry in days configuration.
+ */
+public class PasswordExpiryPolicyUtils {
+    private static final Log log = LogFactory.getLog(PasswordExpiryPolicyUtils.class);
+
+    private PasswordExpiryPolicyUtils(){}
+
+    /**
+     * Get identity event property for password expiry in days.
+     * @param tenantDomain user tenant domain
+     * @param propertyName
+     * @return property value
+     */
+    public static String getIdentityEventProperty(String tenantDomain, String propertyName) {
+
+        // Retrieving properties set in identity event properties
+        String propertyValue = null;
+        try {
+            ModuleConfiguration moduleConfiguration = IdentityEventConfigBuilder.getInstance()
+                    .getModuleConfigurations(PasswordExpiryValidationConstants.PASSWORD_CHANGE_EVENT_HANDLER_NAME);
+
+            if (moduleConfiguration != null) {
+                propertyValue = moduleConfiguration.getModuleProperties().getProperty(propertyName);
+            }
+        } catch (IdentityEventException e) {
+            //if identity property for password expiry in days can not be fetched, default value will be used.
+            log.warn("An error occurred while retrieving module properties");
+            if (log.isDebugEnabled()) {
+                log.debug("An error occurred while retrieving module properties of " +
+                        PasswordExpiryValidationConstants.PASSWORD_CHANGE_EVENT_HANDLER_NAME, e);
+            }
+        }
+        return propertyValue;
+    }
+
+    /**
+     * Get resident IDP property for password expiry in days configuration.
+     * @param tenantDomain user tenant domain
+     * @param propertyName
+     * @return property value
+     */
+    public static String getResidentIdpProperty(String tenantDomain, String propertyName) {
+
+        IdentityProvider residentIdP = null;
+        try {
+            residentIdP = IdentityProviderManager.getInstance().getResidentIdP(tenantDomain);
+        } catch (IdentityProviderManagementException e) {
+        //if the residentIdp property could not be fetched, the identity event property password expiry in days
+            // configuration will be used.
+        }
+
+        if (residentIdP == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Resident IdP is not found for tenant: " + tenantDomain);
+            }
+            return null;
+        }
+        IdentityProviderProperty property = IdentityApplicationManagementUtil
+                .getProperty(residentIdP.getIdpProperties(), propertyName);
+
+        String propertyValue = null;
```

Comment:
Is there any usage of this ? why can't we simply return property.getValue() in line 93 ?